### PR TITLE
#1006: Change 'com.zerocracy.pmo' classes constructor to accept `Pmo`…

### PR DESCRIPTION
**Pull request by proshin-roman**
PR for #1006.

I've changed existing ctors and added new ones where it's possible so that classes receive PMO or Project. A few tests were fixed.

### CODE REVIEW

**Comment by 0crat** [comment]
Job [#1030](https://github.com/zerocracy/farm/issues/1030) is now in scope, role is `REV`

**Comment by 0crat** [comment]
This pull request [#1030](https://github.com/zerocracy/farm/issues/1030) is assigned to @carlosmiranda[/z](https://www.0crat.com/u/carlosmiranda), here is [why](https://www.0crat.com/footprint/C3NDPUA8L/1401195936); the budget is 15 minutes, see [§4](http://www.zerocracy.com/policy.html#4); please, read [§27](http://www.zerocracy.com/policy.html#27) and when you decide to accept the changes, inform @yegor256[/z](https://www.0crat.com/u/yegor256) (the architect) right in this ticket; if you decide that this PR should not be accepted ever, also inform the architect; this [blog post](http://www.yegor256.com/2015/02/09/serious-code-reviewer.html) will help you understand what is expected from a code reviewer

**Comment by 0crat** [comment]
Job [#1030](https://github.com/zerocracy/farm/issues/1030) is already in scope

**Comment by 0crat** [comment]
Job [#1030](https://github.com/zerocracy/farm/issues/1030) is already in scope

**Comment by 0crat** [comment]
Job [#1030](https://github.com/zerocracy/farm/issues/1030) is already in scope

File: src/main/java/com/zerocracy/pmo/People.java [hunk 1]
```diff
@@ -75,6 +75,14 @@ public People(final Farm farm) {
      * @param pkt Project
      */
     public People(final Project pkt) {
```
Comment (carlosmiranda) [inline src/main/java/com/zerocracy/pmo/People.java]: @proshin-roman I think we should remove this constructor altogether. We should pass in `Farm` where it used to pass `Project`.

File: src/main/java/com/zerocracy/pmo/Pmo.java [hunk 2]
```diff
@@ -38,16 +38,34 @@
      */
     private final Scalar<Project> pkt;
 
+    /**
+     * Ctor.
+     * @param project Project
+     */
+    public Pmo(final Project project) {
```
Comment (carlosmiranda) [inline src/main/java/com/zerocracy/pmo/Pmo.java]: @proshin-roman This constructor is unnecessary. We should use the `Pmo(Farm)` constructor.

File: src/main/java/com/zerocracy/pmo/Pmo.java [hunk 3]
```diff
@@ -38,16 +38,34 @@
...
+                () -> farm.find("@id='PMO'").iterator().next()
+            )
         );
     }
 
+    /**
+     * Ctor.
+     * @param orig Project
+     */
+    public Pmo(final Scalar<Project> orig) {
```
Comment (carlosmiranda) [inline src/main/java/com/zerocracy/pmo/Pmo.java]: @proshin-roman ditto... we should be using the `Farm` constructor.

File: src/main/java/com/zerocracy/pmo/Projects.java [hunk 4]
```diff
@@ -53,10 +53,19 @@ public Projects(final Farm farm, final String user) {
 
     /**
      * Ctor.
-     * @param pkt Project
+     * @param pkt The project
      * @param user The user
      */
     public Projects(final Project pkt, final String user) {
```
Comment (carlosmiranda) [inline src/main/java/com/zerocracy/pmo/Projects.java]: @proshin-roman We can make this constructor obsolete, and use the constructor above where `Farm` is passed.

File: src/main/java/com/zerocracy/pmo/Rfps.java [hunk 5]
```diff
@@ -59,9 +59,17 @@ public Rfps(final Farm farm) {
 
     /**
      * Ctor.
-     * @param pkt Project
+     * @param pkt The project
      */
     public Rfps(final Project pkt) {
```
Comment (carlosmiranda) [inline src/main/java/com/zerocracy/pmo/Rfps.java]: @proshin-roman same here, we should remove this constructor altogether and use the constructor that passes `Farm` above.

File: src/main/java/com/zerocracy/pmo/Speed.java [hunk 6]
```diff
@@ -55,10 +55,19 @@ public Speed(final Farm farm, final String user) {
 
     /**
      * Ctor.
-     * @param pkt Project
+     * @param pkt The project
      * @param user The user
      */
     public Speed(final Project pkt, final String user) {
```
Comment (carlosmiranda) [inline src/main/java/com/zerocracy/pmo/Speed.java]: @proshin-roman same here, we should remove this constructor and use the one above that passes `Farm`.

**Comment by 0crat** [comment]
@carlosmiranda[/z](https://www.0crat.com/u/carlosmiranda) this job was assigned to you 5days ago. It will be taken away from you soon, unless you close it, see [§8](http://www.zerocracy.com/policy.html#8). Read [this](http://www.yegor256.com/2014/04/13/no-obligations-principle.html) and [this](http://www.yegor256.com/2014/11/24/principles-of-bug-tracking.html), please.

**Comment by proshin-roman** [comment]
@carlosmiranda Thanks for comments - I've fixed all of them. Please, check the pull request again.

**Comment by yegor256** [comment]
@rultor merge

**Comment by carlosmiranda** [comment]
@yegor256 good to merge

**Comment by g4s8** [comment]
@rultor merge

**Comment by proshin-roman** [comment]
@g4s8 @carlosmiranda goddamn, too many people are committing changes :) I've fixed new compilation errors - please try to merge it again.

**Comment by g4s8** [comment]
@proshin-roman there are some conflicts with master, please fix it. I'll not morge any PR for some time to avoid new conflicts.

**Comment by proshin-roman** [comment]
@g4s8 Thanks! I've fixed the conflicts - let's just wait for CI green light.

**Comment by proshin-roman** [comment]
@g4s8 please, merge the pull request (AppVeyor is failing due to #1041).

**Comment by g4s8** [comment]
@rultor merge

**Comment by 0crat** [comment]
@ypshenychka[/z](https://www.0crat.com/u/ypshenychka) please review this job completed by @carlosmiranda[/z](https://www.0crat.com/u/carlosmiranda), as in [§30](http://www.zerocracy.com/policy.html#30); the job will be fully closed and all payments will be made when the quality review is completed

**Comment by 0crat** [comment]
The job [#1030](https://github.com/zerocracy/farm/issues/1030) is now out of scope

**Comment by 0crat** [comment]
Payment to `ARC` for a closed pull request, as in [§28](http://www.zerocracy.com/policy.html#28): +10 point(s) just awarded to @g4s8[/z](https://www.0crat.com/u/g4s8)

**Comment by ypshenychka** [comment]
@0crat quality good

**Comment by 0crat** [comment]
Order was finished, quality is "good": +20 point(s) just awarded to @carlosmiranda[/z](https://www.0crat.com/u/carlosmiranda)

**Comment by 0crat** [comment]
Quality review completed: +8 point(s) just awarded to @ypshenychka[/z](https://www.0crat.com/u/ypshenychka)

**Pull request merged**